### PR TITLE
@types/catbox: Fix PolicyOptionVariants<T> to always extends DecoratedPolicyOptions<T>

### PR DESCRIPTION
**Pull request by ppenelon**
Hello,

**TL;DR**
This small fix avoid Typescript compilation and type-checking to **always** let `PolicyOptionVariants<T>` be extends as `DecoratedPolicyOptions<T>` ([line 143](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/catbox/index.d.ts#L143)) because the `getDecoratedValue` property of the `DecoratedPolicyOptions<T>` interface is optionnal ([line 234](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/catbox/index.d.ts#L234)).

**DETAILS**
`get(id: Id): Promise<O extends DecoratedPolicyOptions<T> ? DecoratedResult<T> : T | null>;` ([line 143](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/catbox/index.d.ts#L143)) is always returning `DecoratedResult<T>` :

In this context `O` is a generic that must extends `PolicyOptions<T> | DecoratedPolicyOptions<T>` and `DecoratedPolicyOptions<T>` is an interface that extends `PolicyOptions<T>`.

The problem is that the `DecoratedPolicyOptions<T>` interface only have a single property which is optionnal allowing `PolicyOptions<T>` to also extends `DecoratedPolicyOptions<T>`.

This is why the following condition `O extends DecoratedPolicyOptions<T>` is always true because `O` always extends `DecoratedPolicyOptions<T>`.

Authors : @jasonswearingen @AJamesPhillips @saboya

### CODE REVIEW

**Comment by ppenelon** [comment]
Travis build should be fixed with #32498.
Build fail is independant to this PR.

**Comment by davecardwell** [comment]
@ppenelon I fixed the issue with @featherjs/authentication-oauth2 in #32498 which has been merged.

To update my pull request #32497 so Travis CI would try a new build I did the following:

```bash
# Sync my fork with the main DefinitelyTyped repo
git remote add upstream https://github.com/DefinitelyTyped/DefinitelyTyped.git
git fetch upstream
git checkout master
git merge upstream/master

# Rebase my branch onto the updated master
git checkout extra-watch-webpack-plugin
git rebase master

# Overwrite my branch on GitHub to trigger a new CI build on my pull request
git push --force origin extra-watch-webpack-plugin
```

Hopefully that should work for you too!

**Review by saboya — APPROVED** [review]

**Review by jasonswearingen — APPROVED** [review]

**Comment by SimonSchick** [comment]
I think it would be better to make this `: true` as that's a little more clear.

**Comment by ppenelon** [comment]
Could you be more explicit ?
Replace `boolean | undefined` by `true` ?
Can't it be `false` or `undefined` ?
We actually need a non-optionnal property.
Actually I'm not one of the developers of this package, I could have miss something.

**Comment by SimonSchick** [comment]
`getDecoratedValue: true;` yes, I tested this locally and it seems to work fine.

**Comment by ppenelon** [comment]
The authors wrote `@default false` above the property and it was optionnal.
Why would it make more sense to put `getDecoratedValue: true;` ?

**Comment by SimonSchick** [comment]
Because the `true` clearly makes it distinct.

**Comment by AJamesPhillips** [comment]
Good analysis BTW. I think that problem confused me for a while and I gave up on it :)

**Comment by ppenelon** [comment]
@AJamesPhillips Oh I see, in this case, wasn't the interface wrong from the beginning ?
Thanks for the details! :)

**Comment by AJamesPhillips** [comment]
So the original typings were for an optional boolean `getDecoratedValue?: boolean` which is correct and allows you to explicitly say you don't want decorated values using: `getDecoratedValue: false`.  Simon's suggestion doesn't allow for that explicit falseyness but I don't think that really matters.  Also the allowance of an explicit key with value `=== undefined` is not a problem according to the current implementation: https://github.com/hapijs/catbox/blob/bcd5026fd3ffd37427eb973ebcc2dd47488024d2/lib/pending.js#L44
I was just adding to Simon's answer to your question :) 👍
